### ISSUE TITLE
mdc-toolbar util.js ReferenceError: Can't find variable: supportsPassive_

### ISSUE THREAD

**Issue by StephenHesperus** [material-components/material-components-web#1298]
## Bug

I'm using `@material/toolbar` to create a _Vue_ component functioning the same as `@material/toolbar`.
Following the docs and framework examples, here's the snippet that's relevant:

```js
import {MDCToolbarFoundation, util} from '@material/toolbar'

...
registerScrollHandler (handler) {
  window.addEventListener('scroll', handler, util.applyPassive())
},
deregisterScrollHandler (handler) {
  window.removeEventListener('scroll', handler, util.applyPassive())
},
...
```

On Chrome Desktop, there's no error message and everything works fine. But on mobile, it says:
```
[Vue warn]: Error in mounted hook: "ReferenceError: Can't find variable: supportsPassive_"

found in

---> <VMdcToolbar> at C:\Users\User\Documents\vue\src\components\VMdcToolbar.vue
       <VMdcToolbarDemo> at C:\Users\User\Documents\vue\src\components\VMdcToolbarDemo.vue
         <App> at C:\Users\User\Documents\vue\src\App.vue
           <Root>
ReferenceError: Can't find variable: supportsPassive_
```

This in turn causes the flexible behavior of _material toolbar_ not working at all.
The problem is with the `util.applyPassive()` call.
Source code from `mdc-toolbar/util.js`:
```js
let supportsPassive_;

// Determine whether the current browser supports passive event listeners, and if so, use them.
export function applyPassive(globalObj = window, forceRefresh = false) {
  if (supportsPassive_ === undefined || forceRefresh) {
    let isSupported = false;
    try {
      globalObj.document.addEventListener('test', null, {get passive() {
        isSupported = true;
      }});
    } catch (e) { }

    supportsPassive_ = isSupported;
  }

  return supportsPassive_ ? {passive: true} : false;
}
```
After reading the source code, I wrote a function that make the problem go away:
```js
function applyPassive (globalObj = window, forceRefresh = false) {
  let supportsPassive_

  if (supportsPassive_ === undefined || forceRefresh) {
    let isSupported = false
    try {
      globalObj.document.addEventListener('test', null, {get passive () {
        isSupported = true
      }})
    } catch (e) { }

    supportsPassive_ = isSupported
  }

  return supportsPassive_ ? {passive: true} : false
}
```
Basically, I move the line `let supportsPassive_` into the `export` body.

### What MDC-Web Version I'm using?

MDC-Web version: `@material/toolbar: "^0.4.8"`

### What browser(s) is this bug affecting?

Chrome and Safari on iOS 10 on iPhone 7.
Chrome version: 61.0.3163.73
user-agent: `Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_3 like Mac OS X) AppleWebKit/603.3.8 (KHTML, like Gecko) Version/10.0 Mobile/14G60 Safari/602.1`

### What OS are you using?

iOS 10.3.3

### What are the steps to reproduce the bug?

As said above.

### What is the expected behavior?

As said above.

### What is the actual behavior?

As said above.

### Any other information you believe would be useful?

Please ask 😄

**Comment by lynnjepsen** [comment]
supportPassive_ is suppose to be outside the scope of the method so that it's cached when the method is called multiple times.

We can't move it into the method. We're kind of surprised supportPassive_ doesn't work as-is for you're use case. Are you doing some compilation? You could look at any compiled JavaScript and check if supportPassive_ is there.

You might get better Vue support by posting this question on stack overflow. We're not Vue experts so our support is limited.

**Comment by StephenHesperus** [comment]
Thank you very much for your reply, @lynnjepsen .

You reminded me that my project actually uses `babel-loader` to compile es6 code into es5. I tried to put the same `util.js` file under my project, and everything works. So it's a problem with the compiling. Once I added the `@material` directory to `babel-loader`, everything works fine.

Since my project uses `babel-loader`, I need to feed `@material` packages into `babel-loader`, too. Make perfect sense!

Since the problem is not about `material-components-web` and the problem is solved. I will close this issue now.
